packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```
@@ -0,0 +1,440 @@
+    const proofType = credentialRequest.options.proofType
+
+    const keyType: string | undefined = proofTypeKeyTypeMapping[proofType]
+    if (!keyType) {
+      throw new AriesFrameworkError(`No Key Type found for proofType ${proofType}`)
+    }
+
+    const verificationMethod = await findVerificationMethodByKeyType(keyType, issuerDidDocument)
+
+    if (!verificationMethod) {
+      throw new AriesFrameworkError(`Missing verification method for key type ${keyType}`)
+    }
+
+    return verificationMethod.id
+  }
+  /**
+   * Processes an incoming credential - retrieve metadata, retrieve payload and store it in the Indy wallet
+   * @param options the issue credential message wrapped inside this object
+   * @param credentialRecord the credential exchange record for this credential
+   */
+  public async processCredential(
+    agentContext: AgentContext,
+    { credentialRecord, attachment, requestAttachment }: FormatProcessCredentialOptions
+  ): Promise<void> {
+    const credentialAsJson = attachment.getDataAsJson<W3cVerifiableCredential>()
+    const credential = JsonTransformer.fromJSON(credentialAsJson, W3cVerifiableCredential)
+    MessageValidator.validateSync(credential)
+
+    // FIXME: we should verify the signature of the credential here to make sure we can work
+    // with the credential we received.
+    // FIXME: we should do a lot of checks to verify if the credential we received is actually the credential
+    // we requested. We can take an example of the ACA-Py implementation:
+    // https://github.com/hyperledger/aries-cloudagent-python/blob/main/aries_cloudagent/protocols/issue_credential/v2_0/formats/ld_proof/handler.py#L492
+
+    // compare stuff in the proof object of the credential and request...based on aca-py
+
+    const requestAsJson = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+    const request = JsonTransformer.fromJSON(requestAsJson, JsonLdCredential)
+
+    if (Array.isArray(credential.proof)) {
```

same here. Maybe we can look at the ACA-Py implementation?

---

We should just do nothing if the proof is an array. If we don't support this at the moment we should throw an error. This can lead to security issues

---

ditto